### PR TITLE
File.expand_path with actual pwd

### LINKED ISSUE #1699: File.expand_path 'relative/path' should append the current working directory
.MRI
```ruby
::File.expand_path 'chapter-01/index.adoc'
# "/home/guillaume/workspace/opensource/nodebook/chapter-01/index.adoc"
```

.Opal
```js
Opal.const_get_qualified('::', 'File').$expand_path('chapter-01/index.adoc');
// "chapter-01/index.adoc"
```

### PR DESCRIPTION
**Pull request by elia**
Fixes #1699

### CODE REVIEW

File: opal/corelib/file.rb [hunk 1]
```diff
@@ -7,20 +7,28 @@ class File < IO
...
+      absolute = path.start_with?(sep)
       new_parts = []
-      parts[0] = Dir.home if parts.first == '~'
-      parts[0] = Dir.pwd if parts.first == '.'
+
+      unless absolute
+        if parts.first == '~'
+          new_parts << Dir.home
+        else
+          new_parts << Dir.pwd
```
Comment (Mogztter) [inline opal/corelib/file.rb]: As far as I understand C code, I think the `Dir.pwd` is only prepend if the `basedir` argument is `nil`: https://github.com/ruby/ruby/blob/0471781b2a7a40681ccdc051f0239eb19719d79a/file.c#L3414-L3425

And I think `basedir` can be a non absolute path... ?

Comment (elia) [reply opal/corelib/file.rb]: Thanks! I jotted down the code last knight without actually checking the original C 😄

Comment (Mogztter) [reply opal/corelib/file.rb]: > ... without actually checking the original C 

"Great minds think alike." :grinning: 

> I jotted down the code last knight...

"last knight" ?! :horse: :crossed_swords: I think you need some sleep :stuck_out_tongue_winking_eye:

Comment (elia) [reply opal/corelib/file.rb]: > "last knight" ?! 🐴 ⚔️ I think you need some sleep 😜

definitely! 😆

**Comment by Mogztter** [comment]
Thanks @elia 👍

**Pull request merged**